adserver/templates/adserver/advertiser/advertisement-detail.html

```diff
@@ -61,9 +65,11 @@ <h1>{% block heading %}{% trans 'Advertisement: ' %}{{ advertisement.name }}{% e
 </div>
 
 <div class="row mt-5">
+  {% if request.user.is_staff or user_advertiser_role  == "Admin" or user_advertiser_role  == "Manager" %}
```

Comment:
One issue is that it will require a parameter (the advertiser or publisher). It's probably a good idea to do that in a method and that's a good place to cache it too. However, we'll probably still need a template tag of sorts since it accepts a parameter.

Reply:
I added methods and attempted to simplify but we still need templatetags.